components/camera/videosource/webcam.go

```diff
@@ -193,55 +199,232 @@ func makeConstraints(attrs *WebcamAttrs, debug bool, logger golog.Logger) mediad
 	}
 }
 
+// findCamera finds a video device and returns a camera with that video device as the source.
+func findCamera(ctx context.Context, attrs *WebcamAttrs, logger golog.Logger) (camera.Camera, error) {
+	debug := attrs.Debug
+	constraints := makeConstraints(attrs, debug, logger)
+	if attrs.Path != "" {
+		return tryWebcamOpen(ctx, attrs, attrs.Path, false, constraints, logger)
+	}
+
+	source, err := gostream.GetAnyVideoSource(constraints, logger)
+	if err != nil {
+		return nil, errors.Wrap(err, "found no webcams")
+	}
+	return makeCameraFromSource(ctx, source, attrs)
+}
+
+func getLabelFromCamera(cam camera.Camera) (string, error) {
+	src, err := camera.SourceFromCamera(cam)
+	if err != nil {
+		return "", errors.Wrap(err, "cannot get source from camera")
+	}
+	d, err := gostream.DriverFromMediaSource[image.Image, prop.Video](src)
+	if err != nil {
+		return "", errors.Wrap(err, "could not get driver from source")
+	}
+
+	return getLabelFromDriver(d), nil
+}
+
 // NewWebcamSource returns a new source based on a webcam discovered from the given attributes.
 func NewWebcamSource(ctx context.Context, attrs *WebcamAttrs, logger golog.Logger) (camera.Camera, error) {
-	var err error
-
-	debug := attrs.Debug
+	cam, err := findCamera(ctx, attrs, logger)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not find video source for camera")
+	}
 
-	constraints := makeConstraints(attrs, debug, logger)
+	type status int
+	const (
+		monitoring status = 1
```

Comment:
this feels like a boolean and we should just use one

Reply:
Removed.